### PR TITLE
Fix error_resilient mode and add tests.

### PR DESCRIPTION
**Pull request by tdaede**

### CODE REVIEW

**Comment by tdaede** [comment]
It's actually a fix for FrameType::SWITCH.

File: Cargo.toml [hunk 1]
```diff
@@ -62,46 +46,34 @@ num-traits = "0.2"
...
 paste = "0.1"
 noop_proc_macro = "0.2.0"
-serde = { version = "1.0", features = ["derive"], optional = true }
 dav1d-sys = { version = "0.2", optional = true }
 aom-sys = { version = "0.1.3", optional = true }
 scan_fmt = { version = "0.2.3", optional = true }
 ivf = { version = "0.1", path = "ivf/", optional = true }
 rayon = "1.0"
-toml = { version = "0.5", optional = true }
+bincode = { version = "1.1", optional = true }
```
Comment (shssoichiro) [inline Cargo.toml]: I'm thinking this was not intentional--it looks like this PR reverts a lot of changes in Cargo.toml.

File: src/encoder.rs [hunk 2]
```diff
@@ -223,7 +223,7 @@ impl Sequence {
       enable_interintra_compound: false,
       enable_masked_compound: false,
       enable_dual_filter: false,
-      enable_order_hint: !config.still_picture,
```
Comment (lu-zero) [inline src/encoder.rs]: is this part of the fix?

Comment (tdaede) [reply src/encoder.rs]: Whoops no.

**Pull request merged**